modules/xfem/src/userobjects/NodeValueAtXFEMInterface.C

```diff
@@ -7,34 +7,31 @@
 //* Licensed under LGPL 2.1, please see LICENSE for details
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
-#include "PointValueAtXFEMInterface.h"
-#include "MooseMesh.h"
+#include "NodeValueAtXFEMInterface.h"
 #include "MooseVariableFE.h"
 #include "XFEM.h"
-#include "LineSegmentCutSetUserObject.h"
-
-#include "libmesh/mesh_tools.h"
+#include "InterfaceMeshCutUserObjectBase.h"
 #include "libmesh/parallel_algebra.h"
 #include "libmesh/parallel.h"
 
-registerMooseObject("XFEMApp", PointValueAtXFEMInterface);
+registerMooseObject("XFEMApp", NodeValueAtXFEMInterface);
 
 InputParameters
-PointValueAtXFEMInterface::validParams()
+NodeValueAtXFEMInterface::validParams()
 {
   InputParameters params = GeneralUserObject::validParams();
   params.addRequiredParam<VariableName>(
       "variable", "The name of the variable that this UserObject operates on");
   params.addParam<UserObjectName>(
-      "geometric_cut_userobject",
-      "Name of GeometricCutUserObject that provides the points to this UserObject.");
+      "interface_mesh_cut_userobject",
+      "Name of InterfaceMeshCutUserObject that provides nodes location to this UserObject.");
```

Comment:
I forgot the story here. I remember we were planning to use `LevelsetCutUserObject`, but it seems that you've changed your mind?

Reply:
I did discuss this before. I leave the `LevelsetCutUserObject` as another option to be implemented in the future. For now, I could not see any advantage of using that.

Reply:
In this parameter description, I think you should change "nodes location" to "cut locations".